WeakAuras/History.lua

```diff
@@ -0,0 +1,88 @@
+--[[
+This file contains all of the interfaces to the history collection. Indexed by uid of the active display.
+
+History:
+  Table containing historical information about display:
+  source: string indicating where this data came from. Possible values:
+    "user": aura was created by the user via the WeakAuras interface
+    "import": aura was imported via WeakAuras.Import (i.e. pasted string or addon channel transmission)
+    "addon": aura was imported via WeakAuras.RegisterDisplay (i.e. an addon requested the addition)
+  data: snapshot of data as it was at last import, or nil if source == "user", and the user has not chosen to save their changes to history
+  addon: if source == "addon", then this string indicates which addon supplied the last import
+  skippedVersions: record of versions that the user has indicated they don't wish to see update notifications for
+  allowUpdates: if false, then WeakAuras will ignore any addon-sourced update requests
+  lastUpdate: UNIX timestamp of the last time the user completed an import for this display
+--]]
+
+local WeakAuras = WeakAuras
+local history -- history db upvalue
+
+function WeakAuras.LoadHistory(historydb)
+  history = historydb
+  if history.clearOldHistory == nil then
+    history.clearOldHistory = 30
+  end
+  if history.clearOldHistory then
```

Comment:
this will always be true

Reply:
Hardly. We already have precedent for settings in WeakAuras which aren’t normally editable (see the action squelch timer, or debug flag). I figure that whenever we get around to actually providing a settings panel, this will be a find setting to include.